otter/convergence/service.py

```diff
@@ -77,7 +82,8 @@ def update_group_state(group, old_state):
 
 
 def execute_convergence(tenant_id, group_id, log,
-                        get_all_convergence_data=get_all_convergence_data):
+                        get_all_convergence_data=get_all_convergence_data,
+                        plan=plan):
```

Comment:
Non-blocking nitpick: Have you considered using `do` on this function? It seems to be serial and has lot of callbacks in it. It might be easier to read with `do`.